### PR TITLE
Related to 2114: reformat docstrings in the 'handlers' folder

### LINKED ISSUE #2114: Improve doc examples rendering
Currently we are rendering our examples in the docs like this https://pytorch.org/ignite/generated/ignite.engine.engine.Engine.html#ignite.engine.engine.Engine which is not very neat.
What we want to do is to find a better way to make it more good looking, and update our docs examples

image

- Replace "Examples" with some sphinx tag or something else, for example, like scikit-image does:

image

### PR DESCRIPTION
**Pull request by toxa23**
Related to #2114

Description:
part 5 of 7, reformat docstrings in the 'handlers' folder, make sure all examples have "Examples:" header, move "Args:" block to the top of method documentation
Check list:

- [ ] New tests are added (if a new feature is added)
- [X] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -499,8 +497,15 @@ def _check_objects(objs: Mapping, attr: str) -> None:
...
-        Exemples:
+        Args:
+            to_load: a dictionary with objects, e.g. `{"model": model, "optimizer": optimizer, ...}`
+            checkpoint: a dictionary with state_dicts to load, e.g. `{"model": model_state_dict,
+                "optimizer": opt_state_dict}`. If `to_load` contains a single key, then checkpoint can contain directly
+                corresponding state_dict.
+            kwargs: Keyword arguments accepted for `nn.Module.load_state_dict()`. Passing `strict=False` enables
+                the user to load part of the pretrained model (useful for example, in Transfer Learning)
 
+        Examples:
```
Comment (KickItLikeShika) [inline ignite/handlers/checkpoint.py:508]: Bug here with the indentation check https://deploy-preview-2220--pytorch-ignite-preview.netlify.app/generated/ignite.handlers.checkpoint.checkpoint#ignite.handlers.checkpoint.Checkpoint.load_objects

**Review by KickItLikeShika — COMMENTED** [review]
Thanks for the neat work, Anton! one comment and we are good!

File: ignite/handlers/checkpoint.py (hunk 1 above)
Comment (toxa23) [reply ignite/handlers/checkpoint.py:508]: Fixed that. Thank you for spotting the issue!

**Review by KickItLikeShika — APPROVED** [review]
LGTM! Thanks a lot Anton!

**Review by vfdev-5 — APPROVED** [review]

**Pull request merged**